source/fe/fe_q_base.cc

```diff
@@ -632,10 +632,7 @@ FE_Q_Base<dim, spacedim>::get_subface_interpolation_matrix(
          ExcDimensionMismatch(interpolation_matrix.m(),
                               source_fe.n_dofs_per_face(face_no)));
 
-  // see if source is a Q or P element
-  if ((dynamic_cast<const FE_Q_Base<dim, spacedim> *>(&source_fe) != nullptr) ||
-      (dynamic_cast<const FE_SimplexPoly<dim, spacedim> *>(&source_fe) !=
-       nullptr))
+  if (source_fe.has_face_support_points(face_no))
```

Comment:
I think that check is also there to catch the case when `source_fe` might actually be composed of several other finite elements. Do you think it would make sense to add a second check of the form `source_fe.n_components() == this->n_components`?

Reply:
Okey. I've added another assert.